book/layout.md

```diff
@@ -8,152 +8,132 @@ next: styles
-meant to contain text, like `<p>`. Others are meant to organize
-things, like `<div>`. You can tell the two types apart by how they
-position their child elements: text inside a `<p>` is laid out in
-lines, left to right, while containers inside a `<div>` are laid out
-in a vertical column, top to bottom. In web lingo, the text inside a
-`<p>` is laid out with "inline layout", while the containers inside a
-`<div>` are laid out with "block layout".
-
-This means your layout engine has different modes: the `<p>` will be
-laid out with an inline layout mode, while the `<div>` will be laid
-out with a block layout mode. What we've been implementing so far in
-this book is the inline layout mode, so in this chapter we'll
-implement block layout mode.
-
-Second: the layout tree. Layout modes can nest: a web page can have an
-`<html>` in block layout that contains a `<body>` in block layout that
-contains a `<p>` in inline layout. So tree-based layout starts by
-traversing the HTML tree to build a _layout tree_, where each layout
-object in the tree is associated with one of the layout modes.
-
-Layout is very complex, so in this chapter the tree will have a very
-simple structure: it'll contain block layout objects at each branch,
-and inline layout objects at the leaves. For example, it might look
-like this:
-
-![A tree of layout objects. The root is a block for the `<html>`
-    element; it has one child, a block for the `<body>` element; it
-    has three children, each blocks as well (for the `<h1>` and two
-    `<p>` elements); and each of those have a single child, an
-    inline.](/im/layout-modes.png)
-
-Finally: the layout algorithm. Each layout object in the layout tree
-needs to be assigned a size and a position. The general rule is that a
-block, like a paragraph, should take up as much horizontal room as it
-can, and should be tall enough to contain everything inside it. That
-means that a layout objects's width is based on its *parent*'s width,
-while its *height* is based on its *children*'s height:
-
-![The flow of information through layout. Width information flows down
-    the tree, from parent to child, while height information flows up,
-    from child to parent.](/im/layout-order.png)
-
-This suggests a step-by-step approach to layout. First, an element
-computes its position and width, based on its parent. Then it lays out
-its children. Finally, with the children's heights computed, the
-element's height is calculated.
-
-Let's now turn these concepts into code.
-
-::: {.further}
-Formally, computations on a tree like this can be
-described by an [attribute grammar](wiki-atgram). Attribute grammar
-engines analyze dependencies between different attributes to determine
-the right order to calculate each attribute.
-:::
 
-[wiki-atgram]: https://en.wikipedia.org/wiki/Attribute_grammar
+For example, consider a web page with a body that contains a heading
+and three paragraphs:
+
+    <body>
```

Comment:
I think it would be great, and instructive, to make this a real example page, perhaps in an iframe so that can also be loaded on its own. I will try this out in a PR once you've landed this one.

Reply:
This is a great idea!